fix toputils: make it more robust to paths that contain `_tops`

zaz opened the pull request:
Previously, the code was simply splitting paths by the text `_tops/`.

There were two errors:

1. It was not checked first that the path started with `_tops/` (only `_tops`).
2. It would cut off paths that contain `_tops/` because we were taking the first index of the split only, so `_tops/this_is_not_a_tops/test.top` would result in a relpath of `_tops/this_is_not_a` (cut off in the middle of the directory).

Fixes [#9949](https://github.com/QubesOS/qubes-issues/issues/9949).

zaz committed:
fix: only recognize `_tops/`, not `_tops<SOMETHING>/`

Avoids IndexError from `relpath[1]`. Fixes #9949.

marmarek commented on _utils/toputils.py:
```diff
@@ -332,10 +332,10 @@ def toppath(self, path, saltenv=None, verify=True):
-                1
-            ]  # pylint: disable=E1101
+        if relpath.startswith(self.topd_directory + os.sep):  # pylint: disable=E1101
+            relpath = "".join(relpath.split(self.topd_directory + os.sep)[
```
Instead of join, maybe pass `1` as a second parameter to split ("maxsplit") ?

zaz replied:
Perfect! Thank you. Maxsplit as a positional argument is deprecated as I keep seeing from SALT warnings, but yes `maxsplit=1` is perfect. Done!

zaz committed:
fix: recognize paths even if they contain `_tops/`

Previously, `_tops/this_is_not_tops/test.top` would result in a relpath
of `_tops/this_is_not` (cut off in the middle of a directory name).
